packaging/debian/postinst

```diff
@@ -52,20 +51,38 @@ case "$1" in
 	# TODO: should not download binaries if it's only a reconfigure
 	echo "Downloading binaries..."
 	for binary in "storagenode" "storagenode-updater"; do
-	    wget -O /tmp/${binary}.zip ${BINARIES_SERVER}/${binary}_linux_amd64.zip
-	    unzip /tmp/${binary}.zip -d /var/storagenode/
-	    mv /var/storagenode/${binary}_linux_amd64 /var/storagenode/${binary}
+	    tmpfile=$(mktemp /tmp/${binary}.zip.XXXXXX)
+	    wget -O $tmpfile ${BINARIES_SERVER}/${binary}_linux_amd64.zip
+	    unzip $tmpfile -d /var/lib/storagenode
+	    mv /var/lib/storagenode/${binary}_linux_amd64 /var/lib/storagenode/${binary}
 	    # so that we're in PATH easily
-	    ln -s /var/storagenode/${binary} /usr/bin/${binary}
+	    ln -s /var/lib/storagenode/${binary} /usr/bin/${binary}
 	done
-	adduser --system --group --no-create-home --quiet $DAEMON_USER
 
 	if [ "$old_storagedir" != "" ] && [ "$storagedir" != "$old_storagedir" ]; then
 	    echo "New storage directory: ${storagedir} (was ${old_storagedir})"
 	else
 	    echo "Storage directory: ${storagedir}"
 	fi
+
+	# storagenode setup
+	/var/lib/storagenode/storagenode setup \
+	--config-dir "/etc/storagenode" \
+	--identity-dir "$identitydir" \
+	--operator.email "$email" \
+	--operator.wallet "$wallet" \
+	--contact.external-address "$external_addr" \
+	--storage.path "$storagedir" \
+	--storage.allocated-disk-space "$diskspace"
+	
+	# add user and fix ownership
+	adduser --system --group --no-create-home --quiet $DAEMON_USER
```

Comment:
You need a small check before this too, in case the user already exists. It's common to do something like

```sh
if ! getent passwd $DAEMON_USER >/dev/null; then
    adduser --system --group --disabled-login --group --home /var/lib/storagenode --quiet $DAEMON_USER
fi
```